packages/core/src/amazonqFeatureDev/util/files.ts

```diff
@@ -35,19 +35,40 @@ export async function prepareRepoData(
         const zip = new AdmZip()
 
         let totalBytes = 0
-        for (const file of files) {
-            const fileSize = (await vscode.workspace.fs.stat(file.fileUri)).size
-            const isCodeFile = isCodeRelatedFile(file.relativeFilePath)
+        const ignoredExtensionMap = new Map<string, number>()
 
-            // TODO: Follow up with telemetry events to record excluded files extensions
-            if (fileSize >= maxFileSizeBytes || !isCodeFile) {
-                continue
+        await amznTelemetry.amazonq_bundleExtensionIgnored.run(async bundleSpan => {
+            for (const file of files) {
+                const fileSize = (await vscode.workspace.fs.stat(file.fileUri)).size
+                const isCodeFile = isCodeRelatedFile(file.relativeFilePath)
+
+                if (fileSize >= maxFileSizeBytes || !isCodeFile) {
+                    if (!isCodeFile) {
+                        const re = /(?:\.([^.]+))?$/
+                        const extensionArray = re.exec(file.relativeFilePath)
+                        const extension = extensionArray?.length ? extensionArray[1] : undefined
+                        if (extension) {
+                            const currentCount = ignoredExtensionMap.get(extension)
+
+                            ignoredExtensionMap.set(extension, (currentCount ?? 0) + 1)
+                        }
+                    }
+                    continue
+                }
+                totalBytes += fileSize
+
+                const zipFolderPath = path.dirname(file.zipFilePath)
+                zip.addLocalFile(file.fileUri.fsPath, zipFolderPath)
             }
-            totalBytes += fileSize
+            ignoredExtensionMap.forEach((value, key) => {
+                const event = {
+                    filenameExt: key,
+                    count: value,
+                }
 
-            const zipFolderPath = path.dirname(file.zipFilePath)
-            zip.addLocalFile(file.fileUri.fsPath, zipFolderPath)
-        }
+                bundleSpan.record(event)
```

Comment:
FYI: `amazonq_bundleExtensionIgnored.run()` emits one metric. This `bundleSpan.record()` call will overwrite the fields for the "current" metric.

If you want to emit one `amazonq_bundleExtensionIgnored` metric for each filetype, call `amazonq_bundleExtensionIgnored.run()` multiple times (and thus probably want to move the for-loop out of this `run()`).